### ISSUE TITLE
Synthetic data sin_freqs above nyquist frequency

### ISSUE THREAD

**Issue by faniuy** [magenta/ddsp#217]
The sin_freqs features generated by generate_notes_v2 include frequencies higher than nyquist frequency which cause a very high KDE consistency loss ~60 in pretraining.
Is that normal/on purpose?

**Comment by jesseengel** [comment]
hmm, there shouldn't be any loss from those high frequencies because they should have 0 amplitude according to this line: https://github.com/magenta/ddsp/blob/4909fd899c2faa8749e22c8d73148b6d53d08b90/ddsp/training/data_preparation/synthetic_data.py#L263

Are you seeing something different?

**Comment by faniuy** [comment]
I think that line of code only removes base frequencies above nyquist, not harmonic frequencies. Also I found high "sin_freqs" feature in the dataset provided on this repo.

**Comment by jesseengel** [comment]
Yah, that's expected, but their amplitudes should be 0.0 as that line of
code doesn't remove the high frequencies, but sets the amplitude of
anything harmonics above nyquist to 0.0. Please do let me know if you find
sin_freqs with a frequency higher than nyquist and amplitude > 0.0.

**Comment by faniuy** [comment]
i mean, yes, i found sin_freqs higher than nyquist frequency with sin_amps > 0.0 in the provided dataset.
Because the line:
harm_dist = ddsp.core.remove_above_nyquist(f0_hz, harm_dist) 
only removes harm_dist with base freq(not all harmonic freqs) higher than nyquist freq.
I think the code should be:
sin_amps = ddsp.core.remove_above_nyquist(sin_freqs, sin_amps) 

sorry if i'm not clear or wrong cuz im typing on the phone, and im not good at it.

**Comment by jesseengel** [comment]
Sorry for the delay, I got caught up in some conference deadlines and wasn't thinking clearly. I think you're right, which means there's actually a pretty serious bug. It "should" help training to fix it (famous last words), but I should double check with some runs. Things are still busy, so it may take a little bit, sorry for the wait, but thanks for bringing this up and sticking with it.

**Comment by faniuy** [comment]
DDSP is such an awesome work that worth years waiting. :)

I've modified onsets_frames_transcription to transcribe polyphonic(any) music and it worked pretty good and let me see the possibilities. The only problem is that the self-supervising training data I made using DAWs and sound libs can not cover enough sound patterns. So I thought that maybe the best way is to be supervised on spectrograms and propagate back to notes. And then I found that you DDSP papers...

Glad you do it in the fundamental way to really enhance our understanding of music while models like Jukebox treat music like a black box.
Maybe one day all people can communicate in music just like we can talk in words.
